fendermint/vm/interpreter/src/fvm/gas/eip1559.rs

```diff
@@ -0,0 +1,108 @@
+use crate::fvm::gas::{Gas, GasMarket};
+use crate::fvm::state::{read_actor_state, FvmExecState};
+use crate::fvm::FvmMessage;
+use anyhow::anyhow;
+use fendermint_actor_gas_market::EIP1559GasState;
+use fendermint_vm_actor_interface::gas::GAS_ACTOR_ADDR;
+use fendermint_vm_actor_interface::system;
+use fvm_ipld_blockstore::Blockstore;
+use std::sync::atomic::{AtomicU64, Ordering};
+
+type AtomicGas = AtomicU64;
+
+/// The gas market based on EIP1155
+#[derive(Default)]
+pub struct EIP1559GasMarket {
+    /// The block gas limit
+    block_gas_limit: AtomicGas,
+    /// The accumulated gas usage so far
+    block_gas_used: AtomicGas,
+}
+
+impl GasMarket for EIP1559GasMarket {
+    type State = EIP1559GasState;
+
+    fn reload_from_chain<DB: Blockstore + Clone + 'static>(
```

Comment:
This should just be `load`.

Reply:
I moved `load` out of the trait. This method is only needed because of the `&self` trait bound, ideally the gas market should be recreated in `begin`. `GasMarket` does not really need to be aware of self loading from state. I shifted to the implementation of `ActorGasMarket`.